jsapp/js/components/languages/languageSelector.tsx

```diff
@@ -176,27 +176,25 @@ class LanguageSelector extends React.Component<
     }
   }
 
-  fetchSuggestedLanguages() {
-    this.setState({suggestedLanguages: undefined});
+  async fetchSuggestedLanguages() {
+    this.setState({ suggestedLanguages: undefined });
     if (this.props.suggestedLanguages) {
-      this.props.suggestedLanguages.forEach(async (languageCode) => {
-        try {
-          const language = await languagesStore.getLanguage(languageCode);
-          // Just a safe check if suggested languages list didn't change while we
-          // waited for the response.
-          const isAlreadyAdded = Boolean(this.state.suggestedLanguages?.find((stateLanguage) => stateLanguage.code === language.code));
-          if (
-            this.props.suggestedLanguages?.includes(language.code) &&
-            !isAlreadyAdded
-          ) {
-            const newLanguages = this.state.suggestedLanguages || [];
-            newLanguages.push(language);
-            this.setState({suggestedLanguages: newLanguages});
-          }
-        } catch (error) {
-          console.error(`Language ${languageCode} not found 2`);
-        }
-      });
+      try {
+        const languages = await Promise.all(
+          this.props.suggestedLanguages.map(async (languageCode) => {
+            try {
+              return await languagesStore.getLanguage(languageCode);
+            } catch (error) {
+              console.error(`Language ${languageCode} not found 2`);
+              return null;
+            }
+          })
+        );
+        const validLanguages = languages.filter((language) => language !== null);
+        this.setState({ suggestedLanguages: validLanguages });
```

Comment:
I don't see how a new synonym helps more than confuses.
```suggestion
        const suggestedLanguages = languages.filter((language) => language !== null);
        this.setState({ suggestedLanguages });
```